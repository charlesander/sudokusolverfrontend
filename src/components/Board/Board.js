import React, {Component} from 'react';
import PropTypes from 'prop-types'
import Cell from '../Cell/Cell.js'
export default class Board extends Component {

    static propTypes = {
        cellValues: PropTypes.array.isRequired
    }


    render() {

        const items = []


        for (var i = 0; i < 81; i++) {
            items.push(<Cell
                key={i}
                cellIndex={i}
                cellValue={this.props.cellValues[i]}
                setCellValue={this.props.setCellValue}/>)

        }

        return (<div className={"board"}>
            {items}
        </div>)
    }
}


