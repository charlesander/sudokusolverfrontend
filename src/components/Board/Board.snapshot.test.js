import React from "react";
import { create } from "react-test-renderer";
import Board from "./Board"

describe("Board component", () => {
    test("Matches the snapshot", () => {
        let cellValues = [
            0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0, 0, 0, 0]
        let setCellValue = () => true

        const board = create(<Board
            cellValues={cellValues}
            setCellValue={setCellValue}
        />);
        expect(board.toJSON()).toMatchSnapshot();
    });
})

describe("Board component", () => {
    test("Matches the snapshot", () => {
        let cellValues = [
            1, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 2, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 3, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 4, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 5, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 6, 0, 0, 0,
            0, 0, 0, 0, 0, 0, 7, 0, 0,
            0, 0, 0, 0, 0, 0, 0, 8, 0,
            0, 0, 0, 0, 0, 0, 0, 0, 9]
        let setCellValue = () => true

        const board = create(<Board
            cellValues={cellValues}
            setCellValue={setCellValue}
        />);
        expect(board.toJSON()).toMatchSnapshot();
    });
})