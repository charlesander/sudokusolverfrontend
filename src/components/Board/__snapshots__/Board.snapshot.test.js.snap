// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component Matches the snapshot 1`] = `
<div
  className="board"
>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
</div>
`;

exports[`Board component Matches the snapshot 2`] = `
<div
  className="board"
>
  <div
    className="cell preset"
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell preset"
    onClick={[Function]}
  >
    2
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell preset"
    onClick={[Function]}
  >
    3
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell preset"
    onClick={[Function]}
  >
    4
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell preset"
    onClick={[Function]}
  >
    5
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell preset"
    onClick={[Function]}
  >
    6
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell preset"
    onClick={[Function]}
  >
    7
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell preset"
    onClick={[Function]}
  >
    8
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell settable"
    onClick={[Function]}
  >
    0
  </div>
  <div
    className="cell preset"
    onClick={[Function]}
  >
    9
  </div>
</div>
`;
